eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/permission/Permission.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.permission;
+
+/**
+ * Represents a single Android permission.
+ *
+ * @author Gabor Keszthelyi
+ */
+public interface Permission<S extends PermissionStatus>
```

Comment:
`PermissionStatus` is generalized to allow multiple ways to define statuses. It's probably true that within one app we won't use different interpretations, but this 5-way version is still an opinionated version (the fact that it was 4-way originally shows this as well), so I wanted to keep the option to use other versions. The trivial would be the 2-way version which only checks for current `checkSelfPermission()` result, so basically what Android provides. And in some circumstance that might be enough, so no need to handle 5 cases of enum values. In that case, no broadcasting is needed in the app.
(The 5-way version to function best assumes of course that it is used always for a certain permission, to be able to track its status properly. Still that wouldn't mean I think that we should restrict other status usage if it fits better somewhere)

And of course `Permission` is generic because of `PermissionStatus`.